### PR TITLE
Rework routing to add handle and label fields

### PR DESCRIPTION
**Pull request by duncanspumpkin**

### CODE REVIEW

File: src/OpenLoco/Vehicles/CreateVehicle.cpp [hunk 1]
```diff
@@ -655,9 +654,9 @@ namespace OpenLoco::Vehicles
 
     // 0x004B1E77
     // Free orderId?
-    static void sub_4B1E77(const uint16_t orderId)
+    static void sub_4B1E77(const RoutingHandle routing)
     {
-        uint16_t baseOrderId = (orderId & ~(0x3F)) / max_num_routing_steps;
+        const uint16_t baseOrderId = routing.getVehicleRef();
```
Comment (duncanspumpkin) [inline src/OpenLoco/Vehicles/CreateVehicle.cpp:659]: should go through and relabel anything routing related that is currently orderId and make it routingId related.

File: src/OpenLoco/Vehicles/VehicleHead.cpp [hunk 2]
```diff
@@ -1236,7 +1236,7 @@ namespace OpenLoco::Vehicles
                 if (var_0C & Flags0C::manualControl)
                 {
                     auto* vehType2 = train.veh2;
-                    if (vehType2->var_36 != var_36 || vehType2->var_2E != var_2E)
+                    if (vehType2->routingHandle != routingHandle || vehType2->subPosition != subPosition)
```
Comment (duncanspumpkin) [inline src/OpenLoco/Vehicles/VehicleHead.cpp:1239]: vehicle 2 likely represents the destination that the vehicle is trying to get to.

**Comment by duncanspumpkin** [comment]
Ive a follow on PR that moves a bunch of the functions into a manager.

**Review by AaronVanGeffen — APPROVED** [review]
Looks good to me. Replay test passes as well.

**Pull request merged**